### PR TITLE
[eclipse/tycho#937] reproducer for 'does not have an expanded version issue

### PR DESCRIPTION
**Pull request by cdietrich**
[eclipse/tycho#937] reproducer for 'does not have an expanded version issue

### CODE REVIEW

**Comment by cdietrich** [comment]
@laeubi if i run this locally i get
[ERROR] The build could not read 1 project -> [Help 1]
org.apache.maven.project.ProjectBuildingException: Some problems were encountered while processing the POMs:
[ERROR] Unknown packaging: eclipse-plugin @ line 7, column 13
how can i avoid it
and if i run the complete build it does not fail.
so i have no clue what causes the problem :(

**Comment by cdietrich** [comment]
is it possible that it is the use of multiple different versions?

**Comment by laeubi** [comment]
If you run it from commandline make sure to pass `-Dtycho-version=....`

**Comment by laeubi** [comment]
> and if i run the complete build it does not fail.

Have you passed the `-Pits` to maven? But be aware this is running very long. So you can ask surefirey to only execute your test with `-Dtest=dull.class.name`

**Comment by cdietrich** [comment]
`-Dtest=dull.class.name -Pits` wont work, cause not all modules have this test

**Comment by cdietrich** [comment]
with mvn clean install -Pits -Dtest=Tycho937Test -DfailIfNoTests=false
i am getting the same problem about unkown packagig.
gonna configure tycho plugin too

**Comment by laeubi** [comment]
> `-Dtest=dull.class.name -Pits` wont work, cause not all modules have this test

You should execute this in the `tycho-its` folder.

File: tycho-its/projects/tycho937/pom.xml [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+		<tycho-qualifier>yyyyMMdd-HHmm</tycho-qualifier>
+	</properties>
+
+	<build>
+		<plugins>
+			<plugin>
+				<groupId>org.eclipse.tycho</groupId>
+				<artifactId>tycho-packaging-plugin</artifactId>
+				<version>${tycho-version}</version>
+				<configuration>
```
Comment (laeubi) [inline tycho-its/projects/tycho937/pom.xml:20]: I think you need to add the Tycho plugin here not only the packing plugin!

**Comment by cdietrich** [comment]
yes now test is green so it does not reproduce the problem :(

File: tycho-its/projects/tycho937/pom.xml (hunk 1 above)
Comment (cdietrich) [reply tycho-its/projects/tycho937/pom.xml:20]: yes now test is green, so it does not show the problem

**Comment by laeubi** [comment]
> yes now test is green so it does not reproduce the problem :(

The testfailure is there the build succeeds see https://github.com/eclipse/tycho/runs/6302785138

File: tycho-its/projects/tycho937/pom.xml (hunk 1 above)
Comment (cdietrich) [reply tycho-its/projects/tycho937/pom.xml:20]: is it possible jenkins and local(mac) behave differently?

**Comment by cdietrich** [comment]
looks like a jenkins problem
https://ci.eclipse.org/mwe/job/mwe2/job/cd_tycho30stripDown/
is also red but green on my local machine

**Comment by laeubi** [comment]
As mentioned in your example you don't enable tycho and thus the packaging type is unknown.

**Comment by cdietrich** [comment]
@laeubi i can finally reproduce.
updated the pr
javadoc is also involved

**Comment by laeubi** [comment]
Thanks I'll try to take a look at this!

**Comment by laeubi** [comment]
Interestingly this is cause by the **order** of goals:

- `mvn clean javadoc:aggregate-jar verify` -> fails
- `mvn clean verify javadoc:aggregate-jar` -> works

**Comment by cdietrich** [comment]
hmmm since when are goals order dependent

**Comment by laeubi** [comment]
> hmmm since when are goals order dependent

just try `package clean` (will clean your build output after build finished!)

**Comment by laeubi** [comment]
Merged as part of https://github.com/eclipse/tycho/pull/941 thanks for the reproducer

**Pull request closed without merging**